tests/test_downloader_handlers.py

```diff
@@ -372,6 +380,13 @@ def test_download_with_maxsize(self):
         yield self.assertFailure(d, defer.CancelledError, error.ConnectionAborted)
 
     @defer.inlineCallbacks
+    def test_download_with_maxsize_very_large_file(self):
+        request = Request(self.getURL('largechunkedfile'))
+
+        d = self.download_request(request, Spider('foo', download_maxsize=1500))
+        yield self.assertFailure(d, defer.CancelledError, error.ConnectionAborted)
```

Comment:
This test is incomplete at this time. I'm thinking in checking the calls to the logger or there received bytes in the response reader. It shouldn't pass certain threshold which depends on the chunks size.

I think there still a little room for improvement. Twisted keeps a buffer of the data received until the time it can be passed to the client protocol, see:

https://github.com/twisted/twisted/blob/trunk/src/twisted/web/_newclient.py#L1094
https://github.com/twisted/twisted/blob/trunk/src/twisted/web/_newclient.py#L1130

So, if the reactor is busy enough, it can hold several chunks of data which then are passed at once to our response reader causing the repeated log messages.